text/3579-pki.md

```diff
@@ -0,0 +1,309 @@
+
+**Storage:** PKCS #11 Certificates Stored in CloudHSM. These are backed up in-cloud every 24 hours. In the event complete loss of key material occurs, it shall be treated as a full out-of-cycle rotation.
+
+**Expiration:** Root and Top-Level Delegate keys shall follow a `7 year expiration` schedule, except for the first keys of this proposal, which shall have an expiration date of the expected release date of the Rust 2030 edition plus 1 year.
+
+Expiration allows us to re-evaluate algorithm choice and algorithm strength, archive transparency logs, and have a well-tested path for root key replacement.
+
+### First-Issue Expiration Plan
+
+We shall schedule expiration on the key to match the `Rust Edition Cycle`, so any key expiration will also fall within the time frame of a new Rust edition. This cadence is chosen to ease the rotation and release and dissemination of a new `Root Key` aligning with an edition and thus, new releases of `cargo` and associated tooling.
+
+The first key shall be created with an expiration of `2031`, or 8 years, allowing for rotation of the key with the `2030 Edition` of Rust.
+
+All subsequent keys shall be created with a `7 year` expiration, aligning with the date of 2 Rust editions into the future, with 1 additional year allowed for key deployment prior to rotation.
+
+- First Root Key: Generated 2024, 7 year expiration. Expires: 2031, Rotated with: `2030 Edition`
+- Second Root Key: Generated 2030, 7 year expiration. Expires: 2037, Rotated with `2036 Edition`
+- This schedule shall continue until the end of the universe.
+
+### Rust Project Root Key
+
+Shall delegate all trust to Rust Project child certificates. From this point, referenced as `Root Key`.
+
+- This key shall be created in an AWS CloudHSM instance and shall reside exclusively in this instance.
+- It shall not be exported or utilized in any other methods except by quorum (5 of 9) directly within the HSM.
+- It will be managed by quorum authentication, preventing access, usage, or changes to quorum regarding this key except when a quorum authentication occurs.
+
+#### Quorum Model
+
+This HSM instance will be created within the Rust Project AWS account. After initial creation and configuration, the quorum shall be the only entity capable of administering or utilizing the root key in the HSM. Network access to the HSM shall be limited to a PKI bastion [Signing Console][signing-console], which shall only be powered on for scheduled signing events. See the reference-level [Signing Console][signing-console] description for further details on this system.
+
+The root key shall follow a `5-of-9` authentication model for all operations. We consider this a reasonable middle ground of quorum to prevent malicious activity, while allowing for rapid response to an event requiring a quorum. These events are iterated below in [When the Quorum will be needed][when-the-quorum-will-be-needed].
```

Comment:
I think this might need elaboration: is the 5-of-9 scheme here a cryptographic one (i.e. a reconstructed secret key, or threshold signing), or is it an informal one? If the latter, what are the checks (and transparency mechanisms) to prevent access by a malicious or faithless quorum member?

(Similarly, when quorum membership rotates, how do threshold keys/shared secrets/access credentials get demonstrably revoked or burnt for departing members?)

Reply:
That's a good question.  I think they're referring to the quorum ritual in CloudHSM.  One imagines it would be a cryptographic threshold operation due to the token artifacts you can see in the documentation but I'm having trouble finding the exact details.  CloudHSM is a bit like a black box although they have many FIPS certifications.